index.js

```diff
@@ -9,31 +12,113 @@ function ask(questionText) {
+var player = {
+  currentState: currentRoom, 
+  inventory: []
+}
 
-function enterState(newState) {
-  let validTransitions = states[currentState].canChangeTo;
-  if (validTransitions.includes(newState)) {
-    currentState = newState;
+function enterRoom(newRoom) {
+  let validTransitions = rooms[currentRoom].canMoveTo;
+  if (validTransitions.includes(newRoom)) {
+    currentRoom = newRoom;
   } else {
-    throw 'Invalid state transition attempted - from ' + currentState + ' to ' + newState;
+    throw 'Invalid room transition attempted - from ' + currentRoom + ' to ' + newRoom;
   }
-}
 
 start();
 
 async function start() {
-  const welcomeMessage = `182 Main St.
-You are standing on Main Street between Church and South Winooski.
-There is a door here. A keypad sits on the handle.
-On the door is a handwritten sign.`;
-  let answer = await ask(welcomeMessage);
-  console.log('Now write your code to make this work!');
-  process.exit();
+
+  const userPrompt = `\n>_`;
+  let userAnswer = await ask(rooms[currentRoom].roomMessage + userPrompt);
+  let userAnswerLC = userAnswer.toLowerCase(); 
+  while (userAnswerLC != 'exit') { 
+    if ((userAnswerLC == 'gargle') && (room[currentRoom] == 'outside')) {
+      console.log(`Sorry, I don't know how to ` + userAnswerLC + `.` + userPrompt);
+    } else if (userAnswerLC == 'read sign') {
+      console.log(signMsg + userPrompt);    
+    } else if (userAnswerLC == this.exitKey) {
+        enterRoom();
+      // return;  
+    } else if (userAnswerLC == 'exit') { 
+      console.log("Thanks for playing.");
+      process.exit();
+    } 
+    userAnswer = await ask(this.roomMessage + userPrompt); // keeps on asking
+    userAnswerLC = userAnswer.toLowerCase(); // keeps lowercasing the answer 
+  }
+}  // end of main pgm
+
+// functions should be last?
+
+  console.log(rooms[currentRoom].welcomeMessage);
+  //const userPrompt = `\n>_`;
+ // let userAnswerLC = userAnswer.toLowerCase(); 
+  let answer = await ask('_>');
+  while(answer !== 'exit') {
+  if (answer == 'read sign') {
+    console.log(`The sign says "Welcome to Burlington Code Academy! Come on up to the third floor. If the door is locked, use the code 12345."`);
+      //return userPrompt  
+    
+    }
+    else if(currentRoom == 'foyer' || answer == 'take seven days')
+    {
+      console.log ('You have taken the seven days');
+      player.inventory.push('seven days')
+    }
+    else if (answer == 'take sign') {
+      console.log ('Leave the sign alone, moron.')
+    }
+    
+  else if (answer === 'enter code 12345')
+    {
+      console.log('Success! The door opens. You enter the foyer and the door shuts behind you');
+      enterState('foyer');
+      console.log(rooms[currentState].welcomeMessage);
+    }
+    else if (answer ==='open door')
+    {
+      console.log ('Door is locked!');
+    }
+  
+  else { 
+    console.log("Sorry, I don't understand that.");
+   // process.exit();
+  }
+  answer = await ask('_>');
+  }
```

Comment:
I cannot stress enough how important it is to format your code. Do it every single time you change the code, no matter how minor the change.